src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -620,6 +615,28 @@ static StoragePtr create(const StorageFactory::Arguments & args)
                 args.getLocalContext()->checkMergeTreeSettingsConstraints(initial_storage_settings, storage_settings->changes());
             metadata.settings_changes = args.storage_def->settings->ptr();
         }
+
+        auto constraints = metadata.constraints.getConstraints();
+        if (args.query.columns_list && args.query.columns_list->constraints)
+            for (auto & constraint : args.query.columns_list->constraints->children)
+                constraints.push_back(constraint);
+        if (merging_params.mode == MergeTreeData::MergingParams::Collapsing && storage_settings->add_implicit_sign_column_constraint_for_collapsing_engine)
```

Comment:
we need it for `VersionedCollapsing` as well.

Reply:
okay I've added `VersionedCollapsing` there.